src/modules/be-water-smart/be-water-smart.component.ts

```diff
@@ -0,0 +1,686 @@
+  }
+
+  /**
+   * returns an VirtualMeter based on the given id
+   */
+  getVirtualMeter(virtualMeterId: string): VirtualMeter | undefined {
+    var vMeter = this.vMeters.filter(vMeter => vMeter.id === virtualMeterId);
+
+    console.log(vMeter);
+
+    if (vMeter.length === 1) {
+      return vMeter[0];
+    }
+
+    return undefined;
+  }
+
+  /**
+   * calls bws service to retrieve all algorithms
+   */
+  extractAlgorithms(): void {
+    this.extractAlgorithmsInRecord(() => this.bwsService.getAlgorithms());
+  }
+
+  /**
+   * returns an algorithm based on the given name
+   */
+  getAlgorithm(algorithmName: string): Algorithm | undefined {
+    var algorithm = this.algorithms.filter(
+      algorithm => algorithm.name === algorithmName,
+    );
+
+    if (algorithm.length === 1) {
+      return algorithm[0];
+    }
+
+    return undefined;
+  }
+
+  /**
+   * calls bws service to retrieve all trained models
+   */
+  extractModels(): void {
+    this.extractData(() => this.bwsService.getModels(), "MLModels", "models");
+  }
+
+  // ---------- Checkbox Functions ----------
+
+  toggleSelectedMeter(item: any, event: Event, selectedMeters: any): void {
+    const isChecked = (event.target as HTMLInputElement).checked;
+
+    if (isChecked) {
+      selectedMeters.push(item);
+    } else {
+      const index = selectedMeters.findIndex(
+        (meter: {id: any}) => meter.id === item.id,
+      );
+      if (index > -1) {
+        selectedMeters.splice(index, 1); // Remove the item if unchecked
+      }
+    }
+  }
+
+  /**
+   * makes only one trained model selectable at a time for forecasting
+   * @param item variable holding the checkbox information
+   */
+  toggleSelectedModel(item: any) {
+    if (this.selectedModel === item) {
+      this.selectedModel = undefined; // Untick the selected item
+    } else {
+      this.selectedModel = item; // Tick the selected item
+    }
+  }
+
+  // ---------- VirtualMeterList Functions ----------
+
+  /**
+   * creates a new VMeter with an @input name and the id-list of the selected physical meters.
+   * If successful, user gets informed and all global variables get set back.
+   * If failed, user gets informed
+   */
+  addVMeter(selectedMeters: any, selectedMeter: string): void {
+    if (!selectedMeter) {
+      alert("No Name for Virtual Meter!");
+      return;
+    }
+
+    this.bwsService
+      .addVirtualMeterWithId(
+        selectedMeter,
+        this.createSubMeterList(selectedMeters),
+      )
+      .subscribe({
+        next: response => {
+          if (response.hasOwnProperty("virtualMeterId")) {
+            this.extractVMeters();
+            this.selectedPhysicalMeters = [];
+            this.selectedVirtualMeters = [];
+            this.newVMeterName = "";
+            this.newSuperMeterName = "";
+          }
+        },
+        error: error => {
+          console.log(error);
+        },
+      });
+  }
```

Comment:
You have to unsubscribe every subscription, so either use something else (like `firstValueFrom`, as I mentioned in the service) or remember all subscriptions und unsubscribe them in `onDestroy`, but with the new methods, you usually don't need that. Also `Observable`s can easily be converted into signals using `toSignal`